src/averageAges.js

```diff
@@ -55,10 +64,24 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const mothers = people.reduce((mothersArr, person) => {
+    return people.some((child) => child.mother === person.name) ? [...mothersArr, person] : mothersArr;
+  }, []);
+
+  const children = people.filter(({ mother, sex }) => {
+    return people.some(({ name }) => {
+      return mother === name && (onlyWithSon ? sex === 'm' : true);
+    });
+  });
+
+  const sumOfAges = children.reduce((sum, child) => {
+    const mother = mothers.find((person) => person.name === child.mother);
+    const diff = mother ? child.born - mother.born : 0;
+
+    return sum + diff;
+  }, 0) / children.length;
+
+  return sumOfAges;
```

Comment:
The result of these actions is the average value. 
Additionally, you are concatenating differences, not just ages. 
Please rename 'sumOfAges'✨